R/supervised-model-deployment.R

```diff
@@ -164,6 +161,18 @@ SupervisedModelDeployment <- R6Class("SupervisedModelDeployment",
     if ((nchar(self$params$predictedCol) != 0) & (self$params$type != 'multiclass')) {
       self$params$df[[self$params$predictedCol]] <- NULL
     }
+    
+    # Drop columns dropped in develop
+    columnsToKeep <- self$modelInfo$columnNames
+    toDrop <- c(self$params$predictedCol, self$params$grainCol)
+    columnsToKeep <- columnsToKeep[!(columnsToKeep %in% toDrop)]
+    if (all(columnsToKeep %in% names(self$params$df))) {
```

Comment:
Done, with the help of bear spray.

As for single row predictions with NAs when impute = FALSE, I think it's okay if it doesn't work. This isn't specific to single row predictions: If the user specifies that they don't want to impute, but has missing data in every row, then they haven't provided a single row on which we can make predictions and shouldn't be surprised when no predictions are made. Maybe it would be a good idea to provide a helpful error message explaining why no predictions were generated, but I don't think we're responsible for doing any more than that.